apps/emqx_resource/src/emqx_resource_worker.erl

```diff
@@ -982,7 +1004,7 @@ inflight_new(InfltWinSZ, Id, Index) ->
     inflight_append(TableId, {?SIZE_REF, 0}, Id, Index),
     inflight_append(TableId, {?INITIAL_TIME_REF, erlang:system_time()}, Id, Index),
     inflight_append(
-        TableId, {?INITIAL_MONOTONIC_TIME_REF, erlang:monotonic_time(nanosecond)}, Id, Index
+        TableId, {?INITIAL_MONOTONIC_TIME_REF, make_message_ref()}, Id, Index
```

Comment:
Nit: I think it's more semantic/easy to understand if we don't couple it to `make_message_ref/0`, as the key name is pretty indicative of monotonic time, while `make_message_ref` could change to something else.

Reply:
if we change make_message_ref to something else, then we have a problem.